packages/client/src/utils.ts

```diff
@@ -55,5 +55,19 @@ export const decodeToken = (token: string): IDToken => {
 
 export const nowRoundToSec = () => Math.floor(Date.now() / 1000);
 
-export const encodeBase64 = (input: string) => btoa(input);
-export const decodeBase64 = (input: string) => atob(input);
+const replaceNonUrlSafeCharacters = (base64String: string) =>
+  base64String
+    .replace(/\+/g, '-') // Convert '+' to '-'
+    .replace(/\//g, '_') // Convert '/' to '_'
+    .replace(/=+$/g, ''); // Remove ending '='
+const restoreNonUrlSafeCharacters = (base64String: string) =>
+  base64String
+    .replace(/-/g, '+') // Convert '-' to '+'
+    .replace(/_/g, '/'); // Convert '_' to '/'
```

Comment:
nit: the comments is redundant, see [this doc](https://www.cs.utah.edu/~germain/PPS/Topics/commenting.html#:~:text=How%20to%20comment%20Code%3A,not%20%22self%2Ddocumenting%22.).

Reply:
The comments are just for who is unfamiliar with regex.
But I agree with you now, there is little need to explain regex to them.